### ISSUE TITLE
1.x: NullPointerException in OperatorSwitch

### ISSUE THREAD

**Issue by artem-zinnatullin** [ReactiveX/RxJava#4004]
Reported from app version with RxJava 1.1.3, but `OperatorSwitch` was not changed since 1.1.3 so it still looks actual.

``` java
Caused by java.lang.NullPointerException
rx.internal.operators.OperatorSwitch$SwitchSubscriber.drain (OperatorSwitch.java:316)
rx.internal.operators.OperatorSwitch$SwitchSubscriber.childRequested (OperatorSwitch.java:266)
rx.internal.operators.OperatorSwitch$SwitchSubscriber$2.request (OperatorSwitch.java:124)
rx.Subscriber.request (Subscriber.java:157)
rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.call (OperatorObserveOn.java:220)
rx.internal.schedulers.ScheduledAction.run (ScheduledAction.java:55)
```

On the line: [`if (localIndex.get() == inner.id) {`](https://github.com/ReactiveX/RxJava/blob/v1.1.3/src/main/java/rx/internal/operators/OperatorSwitch.java#L316)

`localIndex` is `final`, so looks like `inner` is null.

// JFYI it was on Samsung with Android 6.0.1.

**Comment by akarnokd** [comment]
This appears to be a bug with the JCTools queue. The fix is simply swap two operations in 3 places, but before I post the fix, I'll sort this out with JCTools.

**Comment by akarnokd** [comment]
Fix in #4005.

**Comment by akarnokd** [comment]
Fix delivered in 1.1.6. Thanks!
